src/libraries/Microsoft.Extensions.DependencyModel/src/DependencyContextLoader.cs

```diff
@@ -50,6 +51,7 @@ private static Stream GetResourceStream(Assembly assembly, string name)
             return assembly.GetManifestResourceStream(name);
         }
 
+        [RequiresAssemblyFiles(Message = "Calls GetDepsJsonPath")]
```

Comment:
Public APIs should have some meaningful and actionable message for the attribute.
Granted there's no action in this case (we simply don't support single-file with this API), but at least it should talk in terms of the user's experience and not implementation details.